packages/test/src/test-schedules.ts

```diff
@@ -36,13 +41,65 @@ const calendarSpecDescriptionDefaults: CalendarSpecDescription = {
 };
 
 if (RUN_INTEGRATION_TESTS) {
+  test.before(async () => {
+    const connection = await Connection.connect();
+    try {
+      const client = new Client({ connection }).workflow;
+      // In case we're running with a server that doesn't use the docker-compose setup.
+      try {
+        await connection.operatorService.addSearchAttributes({
+          namespace: 'default',
+          searchAttributes: {
+            CustomIntField: iface.temporal.api.enums.v1.IndexedValueType.INDEXED_VALUE_TYPE_INT,
+            CustomBoolField: iface.temporal.api.enums.v1.IndexedValueType.INDEXED_VALUE_TYPE_BOOL,
+            CustomKeywordField: iface.temporal.api.enums.v1.IndexedValueType.INDEXED_VALUE_TYPE_KEYWORD,
+            CustomTextField: iface.temporal.api.enums.v1.IndexedValueType.INDEXED_VALUE_TYPE_TEXT,
+            CustomDatetimeField: iface.temporal.api.enums.v1.IndexedValueType.INDEXED_VALUE_TYPE_DATETIME,
+            CustomDoubleField: iface.temporal.api.enums.v1.IndexedValueType.INDEXED_VALUE_TYPE_DOUBLE,
+          },
+        });
```

Comment:
Please refactor to separate helper and use in this file as well as integration-tests.ts.
We should be able to remove this retry logic when [this](https://github.com/temporalio/temporal/issues/4017) server issue is fixed.